### PR TITLE
Add best practice for adding a new conditional type and using it internally

### PR DESCRIPTION
**Pull request by prprabhu-ms**
# Why

This pattern has come up a couple times in the last sprint.

# How Tested
Build beta and stable flavor build, verify that preprocessed files match expectations.

### CODE REVIEW

File: packages/acs-ui-common/src/conditional-compilation-sample/index.tsx [hunk 1]
```diff
@@ -204,49 +205,47 @@ export const myExtensibleSelector: MyExtensibleSelector = dummyCreateSelector(
...
+  impl.goodOldTestedFunctionality();
+  if (shouldIncludeUnstableFeature()) {
+    const arg: ThisIsUnstableType = 'getThisFromUser';
+    impl.newMethodForUnstableFunctionality(arg);
+  }
+}
+
+function shouldIncludeUnstableFeature(): boolean {
+  /* @conditional-compile-remove-from(stable) */
+  return true;
```
Comment (PorterNan) [inline packages/acs-ui-common/src/conditional-compilation-sample/index.tsx:249]: Can we double check if ts compiler or linter will throw error for this double return setences?

Comment (prprabhu-ms) [reply packages/acs-ui-common/src/conditional-compilation-sample/index.tsx:249]: I checked that neither does. CI is also green (runs both build and lint).

**Review by JamesBurnside — APPROVED** [review]

File: packages/acs-ui-common/src/conditional-compilation-sample/index.tsx [hunk 2]
```diff
@@ -204,49 +205,47 @@ export const myExtensibleSelector: MyExtensibleSelector = dummyCreateSelector(
...
+ * - Call the new @beta class method from @public API, but only in beta flavored build.
+ *
+ * Following the general guidelines stated above, we achieve this as follows:
+ * - Unconditionally add the new type
+ *   - Conditionally export the new type in the public API.
+ * - Conditionally add the new class method that uses the conditionally exposed type.
+ * - In the @public method, extract a simple helper function that detects whether the @beta method should be called.
+ *   The implementation of this function uses conditional compilation.
+ *   - Call the method if needed. Because the method and type are defined unconditionally, no conditional
+ *     compliation is needed here.
```
Comment (edwardlee-msft) [inline packages/acs-ui-common/src/conditional-compilation-sample/index.tsx:221]: correct compliation to compilation.

Comment (prprabhu-ms) [reply packages/acs-ui-common/src/conditional-compilation-sample/index.tsx:221]: Ouch, auto-submit was enabled.

I can't spell to safe my life :/

**Pull request merged**